### PR TITLE
Add top-level "C" functions for RustBuffer (#1153)

### PR DESCRIPTION
**Pull request by bendk**
This is to help support the desktop JS bindings that we're currently working on.  See #1153 for more details.

### CODE REVIEW

File: uniffi/src/ffi/rustbuffer.rs
Comment (tarikeshaq) [inline uniffi/src/ffi/rustbuffer.rs:204]: nit: should we explicitly link to the destructor? ```[`uniffi_rustbuffer_free`]``` might work

File: uniffi_bindgen/src/scaffolding/templates/RustBuffer.rs
Comment (tarikeshaq) [inline uniffi_bindgen/src/scaffolding/templates/RustBuffer.rs:20]: Should we keep the `Safety` here (maybe change it a little to reflect the scope of the function) - I see we disable it from clippy (reasonably because it's already in the underlying function) - but just in case users get curious and want to audit the generated code, what do you think?

**Review by tarikeshaq — APPROVED** [review]
A few nits, but overall LGTM!

File: uniffi/src/ffi/rustbuffer.rs
Comment (bendk) [reply uniffi/src/ffi/rustbuffer.rs:198]: I think I'm just going to remove that part.  It's not very precise and redundant with the next sentence.

**Comment by bendk** [comment]
@tarikeshaq thanks for the review!  I just pushed some updates based on it, can you double check that I fixed everything?

File: uniffi_bindgen/src/scaffolding/templates/RustBuffer.rs
Comment (tarikeshaq) [reply uniffi_bindgen/src/scaffolding/templates/RustBuffer.rs:20]: That works! We might want to link to a GitHub permalink, the `docs.rs` or something so users don't have to go looking where `uniffi/src` is

**Comment by tarikeshaq** [comment]
Looks good to me @bendk! Had a tiny comment on referencing the rustbuffer module - but don't feel like you need to block on it (or other reviews from me 😄)

**Pull request merged**
